### ISSUE TITLE
Wrong orientation pointcloud drawing when using cloudviewer - showCloud()

### ISSUE THREAD

**Issue by evertale888** [PointCloudLibrary/pcl#5885]
Hello, i have successfully draw pointcloud using using depth data from rgb-d camera. The data format is a common one (x y z with r g b). I used cloudviewer-showCloud for visualization. The issue is that the draw pointcloud is mirror image to the actual pointcloud image (in upside down). How can i change it so that it will draw in the correct orientation.

**Comment by mvieth** [comment]
Sounds like the problem is most likely that your data follows the left hand rule and PCL follows the right hand rule (or vice versa). How exactly do you create the point cloud? Have you tried inverting one of the three axes? (e.g. `x *= -1`)

**Comment by mishima2323** [comment]
> PCL follows the right hand rule

I created the pointcloud from depth & color data, get their poses then convert them to SE3Quat, then i transform it into pcl format like this: 
pcl::transformPointCloud(*pointCloud_in, *pointCloud_out, poses.inverse().matrix());

Tried inverting one of the axis though. Inverting y will correct it but it only plot a correct pointcloud  just for the early camera frame. As i plot the pointcloud frame by frame from the dataset. The later frame will superimpose the earlier plot. It became a layer of mess haha. Can we change anything from the pcl side?

**Comment by evertale888** [comment]
> Sounds like the problem is most likely that your data follows the left hand rule and PCL follows the right hand rule (or vice versa). How exactly do you create the point cloud? Have you tried inverting one of the three axes? (e.g. `x *= -1`)

I created the pointcloud from depth & color data, get their poses then convert them to SE3Quat, then i transform it into pcl format like this:
pcl::transformPointCloud(*pointCloud_in, *pointCloud_out, poses.inverse().matrix());

Tried inverting one of the axis though. Inverting y will correct it but it only plot a correct pointcloud just for the early camera frame. As i plot the pointcloud frame by frame from the dataset. The later frame will superimpose the earlier plot. It became a layer of mess haha. Can we change anything from the pcl side?
